### PR TITLE
settings store

### PR DESCRIPTION
**Pull request by ixv**
Draft PR for the settings store. Still need to do frontend stuff.

Can someone tell me if I have the pattern right for scrying? I wasn't sure if I should return `~` or `[~ ~]` when scrying for a thing that doesn't exist, but might exist in the future.

### CODE REVIEW

**Comment by liam-fitzgerald** [comment]
Yeah in userspace you always want to use `[~ ~]`. The way I remember it is that ~ is "I don't know anything about this path" and and [~ ~] is I know that I don't have it. ~ is really only useful for scrying foreign ships in kernelspace, as I understand it.

File: pkg/arvo/app/settings.hoon [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  ++  on-watch
+    |=  pax=path
+    ^-  (quip card _this)
+    ?>  (team:title our.bol src.bol)
+    ?+  pax  (on-watch:def pax)
+        [%all ~]
+      [~ this]
+    ::
+        [%bucket @ ~]
+      =/  bucket-key  i.t.pax
```
Comment (liam-fitzgerald) [inline pkg/arvo/app/settings.hoon:62]: Should use tistar instead

**Review by liam-fitzgerald — APPROVED** [review]
Minor quibble, looks good otherwise

**Comment by matildepark** [comment]
ping @tacryt-socryp this could also use a review but later in the week is fine

File: pkg/arvo/app/settings-store.hoon [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      ``settings-data+!>(entry+u.entry)
+    ==
+  ::
+  ++  on-agent  on-agent:def
+  ++  on-leave  on-leave:def
+  ++  on-arvo   on-arvo:def
+  ++  on-fail   on-fail:def
+  --
+::
+|_  bol=bowl:gall
```
Comment (tacryt-socryp) [inline pkg/arvo/app/settings-store.hoon:102]: I generally prefer shadowing `=bowl:gall` so as not to be able to accidentally reference the bunted bowl.

**Review by tacryt-socryp — APPROVED** [review]
LGTM, only one note on bowl shadowing vs `bol`.

**Pull request merged**